fmt/src/formatter.rs

```diff
@@ -2856,6 +2845,242 @@ impl<'a, W: Write> Visitor for Formatter<'a, W> {
         }
         Ok(())
     }
+
+    fn visit_assembly(
+        &mut self,
+        loc: Loc,
+        dialect: &mut Option<StringLiteral>,
+        block: &mut YulBlock,
+        flags: &mut Option<Vec<StringLiteral>>,
+    ) -> Result<(), Self::Error> {
+        write_chunk!(self, loc.start(), "assembly")?;
+        if let Some(StringLiteral { loc, string, .. }) = dialect {
+            write_chunk!(self, loc.start(), loc.end(), "\"{string}\"")?;
+        }
+        if let Some(flags) = flags {
+            if !flags.is_empty() {
+                let loc_start = flags.first().unwrap().loc.start();
+                self.surrounded(loc_start, "(", ")", Some(block.loc.start()), |fmt, _| {
+                    let mut flags = flags.iter_mut().peekable();
+                    let mut chunks = vec![];
+                    while let Some(flag) = flags.next() {
+                        let next_byte_offset = flags.peek().map(|next_flag| next_flag.loc.start());
+                        chunks.push(fmt.chunked(flag.loc.start(), next_byte_offset, |fmt| {
+                            write!(fmt.buf(), "\"{}\"", flag.string)?;
+                            Ok(())
+                        })?);
+                    }
+                    fmt.write_chunks_separated(&chunks, ",", false)?;
+                    Ok(())
+                })?;
+            }
+        }
+
+        block.visit(self)
+    }
+
+    fn visit_yul_block(
+        &mut self,
+        loc: Loc,
+        statements: &mut Vec<YulStatement>,
+        attempt_single_line: bool,
+    ) -> Result<(), Self::Error> {
+        if attempt_single_line && statements.len() == 1 {
+            let chunk = self.chunked(loc.start(), Some(loc.end()), |fmt| {
```

Comment:
i'm worried because we're not doing this in a temporary buffer, this might alter the comments. e.g.

```
assembly { statement /* a really long comment */ }
```

Reply:
just wrote a test for this, if the comment exceeds the max line length, then it ends up multilined. the reason for that is that the block is treated like a chunk and it wouldn't fit on one line